doc/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2024-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+find_package(Doxygen COMPONENTS dot)
+
+if(DOXYGEN_FOUND)
+  set(doxyfile ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile)
+  configure_file(Doxyfile.in ${doxyfile})
+
+  # In CMake 3.27, The FindDoxygen module's doxygen_add_docs()
+  # command gained a CONFIG_FILE option to specify a custom doxygen
+  # configuration file.
+  # TODO: Consider using it.
+  set(stampfile ${CMAKE_CURRENT_BINARY_DIR}/.stamp)
+  add_custom_command(
+    OUTPUT ${stampfile}
+    COMMAND Doxygen::doxygen ${doxyfile}
```

Comment:
This "hangs" for a few minutes without producing any output, while doxygen is running and then prints all of its output at once. I found that adding `USES_TERMINAL` makes it output the doxygen progress as it is running.

Reply:
The [`USES_TERMINAL`](https://cmake.org/cmake/help/latest/command/add_custom_command.html) option does not guarantee the same behaviour in all cases.

Reply:
What do you mean? I tested with both `Ninja` and `Unix Makefiles` and it behaves the same - output printed on screen "in real time".

Reply:
[Implemented](https://github.com/hebasto/bitcoin/pull/165#issuecomment-2080404349).